zeyple/zeyple.py

```diff
@@ -24,24 +26,39 @@
 class Zeyple:
     """Zeyple Encrypts Your Precious Log Emails"""
 
-    def __init__(self):
-        self._load_configuration()
+    def __init__(self, config_fname='zeyple.conf'):
+        self.config = self.load_configuration(config_fname)
 
-        log_file = self._config.get('zeyple', 'log_file')
+        log_file = self.config.get('zeyple', 'log_file')
         logging.basicConfig(
             filename=log_file, level=logging.DEBUG,
             format='%(asctime)s %(process)s %(levelname)s %(message)s'
         )
         logging.info("Zeyple ready to encrypt outgoing emails")
 
-        # tells gpgme.Context() where are the keys
-        os.environ['GNUPGHOME'] = self._config.get('gpg', 'home')
+    @property
+    def gpg(self):
+        protocol = gpgme.PROTOCOL_OpenPGP
 
-    def process_message(self, message, recipients):
+        if self.config.has_option('gpg', 'executable'):
+            executable = self.config.get('gpg', 'executable')
+        else:
+            executable = None  # Default value
+
+        home_dir = self.config.get('gpg', 'home')
+
+        ctx = gpgme.Context()
+        ctx.set_engine_info(protocol, executable, home_dir)
+        ctx.armor = True
+
+        return ctx
+
+    def process_message(self, message_data, recipients):
         """Encrypts the message with recipient keys"""
 
-        message = email.message_from_string(message)
-        logging.info("Processing outgoing message %s", message['Message-id'])
+        in_message = email.message_from_string(message_data)
+        logging.info(
+            "Processing outgoing message %s", in_message['Message-id'])
```

Comment:
The line is too long for 80 characters. That's the way you're supposed to break a long line. [Hanging indents *may* be indented to other than 4 spaces.](https://www.python.org/dev/peps/pep-0008/#indentation).